### PR TITLE
add conv search sentiment api connect

### PR DESCRIPTION
**Pull request by janstoff**

### CODE REVIEW

File: src/conversational-search-interactions-api.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const RESPONSE_SERVER_ERROR = require('./api').RESPONSE_SERVER_ERROR;
+require('es6-promise').polyfill();
+
+const conversationalSearchInteractionsInstance =
+  require('./api').conversationalSearchInteractionsInstance;
+
+var putSentimentClick = function (apiHostname, sitekey, conversationId, sentimentValue) {
+  return new Promise((resolve, reject) => {
+    conversationalSearchInteractionsInstance
+      .put(`https://${apiHostname}/v2/indices/${sitekey}/conversations/${conversationId}/rating`, {
```
Comment (haoAddsearch) [inline src/conversational-search-interactions-api.js:11]: I think sending small analytics data should be done with [sendBeacon](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon). Backend endpoint will need to use POST for beacon.

Another point to clarify: how does backend handle this case when user clicks thumbs-up -> un-click -> then click again? Does the whole thing count as 1 or 2 "thumbs-up"? If 1, I'd suggest sending `sessionId` so backend can handle the calculating logic.

Check out this endpoint https://api.addsearch.com/v1/stats/2c32bf3eb06b30af5f8208481aea3e8b/ when searching in https://demo.addsearch.com/addsearch-ready-made-solutions/widget.html

File: src/index.js [hunk 2]
```diff
@@ -95,6 +97,27 @@ var client = function (sitekey, privatekey) {
...
+  };
+
+  /**
+   * Post a copy click value to the conversational search interactions API
+   *
+   * @param {string} conversationId  Argument 1: Conversation ID
+   * @returns {Promise<boolean|Object>} Resolves to true if successful, rejects with an error object if unsuccessful
+   */
+  this.putCopyClick = function (conversationId) {
+    return putCopyClick(this.apiHostname, this.sitekey, conversationId);
```
Comment (haoAddsearch) [inline src/index.js:118]: `putCopyClick` is not imported yet, maybe add TODO to remind ourselves later

File: src/conversational-search-interactions-api.js (hunk 1 above)
Comment (janstoff) [reply src/conversational-search-interactions-api.js:11]: Sentiment is more like a feature of converstational search. Maybe in the end we decide to send a stats event as well in parallel. But I think at it's core it is a direct interaction with the feature.

As to the value sent: The frontend decides which value to pass to putSentimentClick. If nothing is selected or the other value is selected at the moment the respective value is sent, if something is selected that was already selected the value becomes "neutral" i.e. un-select.

File: src/index.js (hunk 2 above)
Comment (janstoff) [reply src/index.js:118]: jup, I think I will just remove for now

File: src/conversational-search-interactions-api.js (hunk 1 above)
Comment (haoAddsearch) [reply src/conversational-search-interactions-api.js:11]: Makes sense, I'll approve this PR.

**Pull request merged**